Sign artifacts conditionally

As pointed out by @fknittel in [this comment](https://github.com/junit-pioneer/junit-pioneer/issues/480#issuecomment-839670979), JUnit Pioneer currently can't be used via JitPack. This is because `publishToMavenLocal` fails as it tries to sign the artifacts ([JitPack log](https://jitpack.io/com/github/junit-pioneer/junit-pioneer/v1.4.1/build.log) on v1.4.1):

```text
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':signMavenPublication'.
> Cannot perform signing task ':signMavenPublication' because it has no configured signatory
```

According to the [Signing Plugin docs](https://docs.gradle.org/current/userguide/signing_plugin.html#sec:conditional_signing), signin artifacts conditionally is a common usage pattern. We should do this too, not just to enable JitPack usage, but also to e.g. allow users to conveniently publish to their local Maven repo.

--- a/README.md
+++ b/README.md
@@ -111,7 +111,7 @@ The least we can do is to thank them and list some of their accomplishments here
 #### 2021
 
 * [Cory Thomas](https://github.com/dump247) contributed the `minSuccess` attribute in [retrying tests](https://junit-pioneer.org/docs/retrying-test/) (#408 / #430)
-* [Daniel Kraus](https://github.com/beatngu13) fixed bugs in the environment variable and system property extensions (#432 / #433, #448 / #449, and #473 / #476)
+* [Daniel Kraus](https://github.com/beatngu13) fixed bugs in the environment variable and system property extensions (#432 / #433, #448 / #449, and #473 / #476) and improved the build process (#482 / #483)
 * [Slawomir Jaranowski](https://github.com/slawekjaranowski) Migrate to new Shipkit plugins (#410 / #419)
 * [Stefano Cordio](https://github.com/scordio) contributed [the Cartesian Enum source](https://junit-pioneer.org/docs/cartesian-product/#cartesianenumsource) (#379 / #409 and #414 / #453)
 

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -174,7 +174,7 @@ publishing {
 
 signing {
 	setRequired({
-		project.version != "unspecified"
+		project.version != "unspecified" && gradle.taskGraph.hasTask("publishToSonatype")
 	})
 	val signingKey: String? by project
 	val signingPassword: String? by project

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -173,6 +173,9 @@ publishing {
 }
 
 signing {
+	setRequired({
+		project.version != "unspecified"
+	})
 	val signingKey: String? by project
 	val signingPassword: String? by project
 	useInMemoryPgpKeys(signingKey, signingPassword)

